### PR TITLE
chore(relay): Check in project config invalidation script

### PR DESCRIPTION
**Pull request by jjbayer**
During incident [INC-288](https://getsentry.atlassian.net/browse/INC-288) ([Notion Doc](https://www.notion.so/sentry/INC-288-Relay-dropping-envelopes-due-to-invalid-project-configs-415225c46b4045f1b3d9982a7e9be2df)), where unparsable project configs had made it into Redis, we weren't able to come up with a script that would invalidate all project configs. It turned out that we have a [function](https://github.com/getsentry/sentry/blob/db40a837cc4e9e17fb3c66d878cc3a0f5b155bff/src/sentry/killswitches.py#L25) that does that, but it is only called from one of the killswitches. This PR converts the same invalidation logic into a stand-alone script.

Note that the script has not been tested in production, so there is still a certain risk involved if we ever need to call it. To reduce this risk, we can test the script with the `--percentage` argument to run it on a subset of orgs.

Fixes https://github.com/getsentry/team-ingest/issues/59.

[INC-288]: https://getsentry.atlassian.net/browse/INC-288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bin/invalidate-project-configs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    configure()
+    from sentry.models import Organization
+    from sentry.tasks.relay import schedule_invalidate_project_config
+    from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+    # NOTE: Instead of iterating over all projects, we could use redis.scan_iter()
+    # to evict only project configs that exist in redis.
+    queryset = Organization.objects.all().values_list("id", flat=True)
+    print("Invalidating all project configs...")  # NOQA S002
+    for org_id in RangeQuerySetWrapperWithProgressBar(queryset, result_value_getter=lambda x: x):
```
Comment (jjbayer) [inline bin/invalidate-project-configs:20]: This fetches all organizations in batches of 1000, which is the slow-but-stable way of iterating that we also use in migrations.

**Review by iker-barriocanal — APPROVED** [review]
Do we maitain this script? Should we add this file to the codeowners file?

**Review by olksdr — APPROVED** [review]

**Comment by mwarkentin** [comment]
> Note that the script has not been tested in production, so there is still a certain risk involved if we ever need to call it.

Would it make sense to have a dry run mode that we could use to at least exercise everything but the actual invalidation? 

Alternatively, would it be safe to run this in production somewhere (maybe during a lower-traffic period)? Would be nice to have some confirmation that this works as expected in a controlled setting rather than during the next incident. :)

**Comment by jjbayer** [comment]
> Would it make sense to have a dry run mode that we could use to at least exercise everything but the actual invalidation?

The dry-run would come down to a `RangeQuerySetWrapper` loop that schedules a dummy Celery task. Would that be helpful? In general, do you think it would be a problem to schedule so many Celery tasks at once?

> Alternatively, would it be safe to run this in production somewhere (maybe during a lower-traffic period)? Would be nice to have some confirmation that this works as expected in a controlled setting rather than during the next incident. :)

Yes, this sounds like a good idea. Instead of a dry run, I could add an option to run this for a percentage or slice of orgs. Then we could test it in production with 1%, 10%, 25%, ... etc. @mwarkentin what do you think?

**Comment by mwarkentin** [comment]
@jjbayer Yes, I think the 2nd option sounds better. :)

**Pull request merged**